OpenAI.SDK/OpenAiOptions.cs

```diff
@@ -1,9 +1,50 @@
 ﻿namespace OpenAI.GPT3
 {
+    /// <summary>
+    ///     Provider Type
+    /// </summary>
+    public enum ProviderType
+    {
+        /// <summary>
+        ///     OpenAi Provider
+        /// </summary>
+        OpenAi = 1,
+
+        /// <summary>
+        ///     Azure Provider
+        /// </summary>
+        Azure = 2
+    }
+
     public class OpenAiOptions
     {
+        private const string OpenAiDefaultApiVersion = "v1";
+        private const string OpenAiDefaultBaseDomain = "https://api.openai.com/";
+        private const string AzureOpenAiDefaultApiVersion = "2022-12-01";
+
+
+        /// <summary>
+        ///     Setting key for Json Setting Bindings
+        /// </summary>
         public static readonly string SettingKey = "OpenAIServiceOptions";
 
+        private string? _apiVersion;
+        private string? _baseDomain;
+        private ProviderType? _providerType;
+
+        /// <summary>
+        ///     Get Provider Type
+        /// </summary>
+        public ProviderType ProviderType
+        {
+            get
+            {
+                _providerType ??= !string.IsNullOrEmpty(DeploymentId) || !string.IsNullOrEmpty(ResourceName) ? ProviderType.Azure : ProviderType.OpenAi;
```

Comment:
Same comment as below, I'd rather just make people be explicit that try and magic my way into things.